.github/workflows/caravel_cocotb.yml

```diff
@@ -0,0 +1,51 @@
+name: Caravel Cocotb CI
+
+on:
+  push:
+    branches:
+      - main  
+      - dev
+      - CI_dev
+    paths:
+      - 'cocotb/caravel_cocotb/CI/**'
+      - '.github/workflows/caravel_cocotb.yml'
+      - 'cocotb/caravel_cocotb/scripts/**'
+      - '!cocotb/caravel_cocotb/**/'  # Exclude directories within the specified directory
+
+
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```

Comment:
You can create a matrix where it tries out several python versions. (This is a future improvement)